Refresh social connections overview page (#3236)
clerkinc/clerk-docs: docs/guides/configure/auth-strategies/social-connections/overview.mdx

---
title: Social connections (OAuth)
description: Learn how to effortlessly add social connections to your application using popular social providers like Google, Facebook, GitHub and more.

> sdk: nextjs, react, astro, expressjs, react-router, tanstack-react-start, ios, android, expo, vue, nuxt, chrome-extension, fastify, js-frontend

Author: Would we not want to include go and ruby too, as otherwise, this is how the sidebar looks for them? It's still got the social connections stuff but without the overview. 

<img width="209" height="128" alt="Image" src="https://github.com/user-attachments/assets/33d27e7b-4c8a-4b3a-a089-4c863eb88b85" />

Reviewer: I brought in Ruby and Go, then took a backend pass through the guide. https://github.com/clerk/clerk-docs/pull/3236/commits/b07c4564764699a1c3ed6fa8ed8c7e83b7da35b5

LMK what you think.

      link: "/docs/getting-started/quickstart/setup-clerk",
      icon: "clerk",
    },
    {
      title: "Follow the quickstart guide.",
      link: "/docs/getting-started/quickstart",
      icon: "clerk",
    },
  ]}
/>

Social connections, also known as OAuth connections in Clerk, allow users to gain access to your application by using their existing credentials from an Identity Provider (IdP), like Google or Microsoft. For example, if you enable Google as a social provider, then when a user wants to sign in to your application, they can select Google and use their Google account to sign in.

> [!NOTE]
> When using social connections, the sign-up and sign-in flows are equivalent. If a user doesn't have an account and tries to sign in, an account will be made for them, and vice versa.

<If notSdk={["ios", "android"]}>
  The easiest way to add social connections to your Clerk app is by using [prebuilt components](/docs/reference/components/overview). If you need more control, you can [build a custom OAuth flow](/docs/guides/development/custom-flows/authentication/oauth-connections) with the Clerk API.
</If>

<If sdk={["ios", "android"]}>

>   The easiest way to add social connections to your Clerk app is by using [prebuilt views](/docs/reference/views/overview). If you need more control, you can [build a custom OAuth flow](/docs/guides/development/custom-flows/authentication/oauth-connections) with the Clerk SDK.

Author: ```suggestion
  The easiest way to add social connections to your Clerk app is by using [prebuilt views](/docs/reference/views/overview). If you need more control, you can [build a custom OAuth flow](/docs/guides/development/custom-flows/authentication/oauth-connections) with the Clerk API.
```

Should this be API to be consistent with the text above? @manovotny

Reviewer: Good catch! It was intentional. We were mirroring "prebuilt components ↔ Clerk API" with "prebuilt views ↔ Clerk SDK" since native devs typically import a platform SDK. But the destination page uses "Clerk API" universally and native flows still call the API through the SDK, so the distinction was more semantic than useful. Updated. Also caught and fixed the same wording in the post-auth-manage block at L455 for consistency.

To add additional OAuth scopes, when you are [enabling a new social connection](#enable-a-social-connection), enable **Use custom credentials**. The **Scopes** field will appear.

<If notSdk={["ios", "android"]}>
  ## Request additional OAuth scopes after sign-up

  Clerk allows you to request additional OAuth scopes even after a user has signed up.

  Pass the [`additionalOAuthScopes`](/docs/reference/components/user/user-profile) prop to the [`<UserProfile/>`](/docs/reference/components/user/user-profile) or [`<UserButton />`](/docs/reference/components/user/user-button) component, with any additional OAuth scope you would like per provider. The user will be prompted to reconnect their account on their user profile page.

>   Use the following tabs to see how to add additional OAuth scopes to the `<UserProfile/>` and `<UserButton/>` components.

Author: I think we should add this sentence to the `Request additional OAuth scopes after sign-up` section for iOS below too. @manovotny

Reviewer: Added — picked up alongside https://github.com/clerk/clerk-docs/pull/3236#discussion_r3148767301.

  <CodeBlockTabs options={["<UserProfile />", "<UserButton />"]}>
    ```tsx {{ filename: 'app/page.tsx' }}
    <UserProfile
      additionalOAuthScopes={{
        google: ['foo', 'bar'],
        github: ['qux'],
      }}
    />
    ```

    ```tsx {{ filename: 'app/page.tsx' }}
    <UserButton
      userProfileProps={{
        additionalOAuthScopes: {
          google: ['foo', 'bar'],
          github: ['qux'],
        },
      }}
    />
    ```
  </CodeBlockTabs>
</If>

<If sdk={["ios"]}>
  ## Request additional OAuth scopes after sign-up

  Clerk allows you to request additional OAuth scopes even after a user has signed up.

>   Apply the `.userProfileOAuthConfig(_:)` modifier to a `UserProfileView` or `UserButton` with any additional OAuth scopes you would like per provider. The user will be prompted to reconnect their account when they open their user profile.

Author: ```suggestion
  Apply the `.userProfileOAuthConfig(_:)` modifier to the `UserProfileView` or `UserButton` view, with any additional OAuth scopes you would like per provider. The user will be prompted to reconnect their account on their user profile page.
```

To be consistent with the following paragraph for non-iOS/Android SDKs:

```
Pass the [`additionalOAuthScopes`](/docs/reference/components/user/user-profile) prop to the [`<UserProfile/>`](/docs/reference/components/user/user-profile) or [`<UserButton />`](/docs/reference/components/user/user-button) component, with any additional OAuth scope you would like per provider. The user will be prompted to reconnect their account on their user profile page.
```

Also should we link  `UserProfileView` and `UserButton` like we linked the components?

Reviewer: Done — reworked the paragraph to match the web pattern ("Apply ... to the `UserProfileView` or `UserButton` view, with..."), linked both views to their reference pages. Skipped linking `.userProfileOAuthConfig(_:)` since there is no in-docs target for it (the iOS API ref lives on `clerk.github.io/clerk-ios`) — happy to revisit if we want to point at an external link.

<If notSdk={["ios", "android"]}>
  If you use [prebuilt components](/docs/reference/components/overview), the [`<UserProfile />`](/docs/reference/components/user/user-profile) component lets users manage their connections. For a custom UI, [build a custom OAuth flow](/docs/guides/development/custom-flows/authentication/oauth-connections) with the Clerk API.
</If>

<If sdk={["ios", "android"]}>
  If you use [prebuilt views](/docs/reference/views/overview), the [`UserProfileView`](/docs/reference/views/user/user-profile-view) lets users manage their connections. For a custom UI, [build a custom OAuth flow](/docs/guides/development/custom-flows/authentication/oauth-connections) with the Clerk SDK.
</If>

<If sdk={["ios", "android"]}>
  ## Allowlist OAuth redirect URLs

  In addition to enabling a social connection in the Clerk Dashboard, native applications require allowlisting the redirect URLs used during the OAuth flow.

  <Include src="_partials/allowlist-redirect-urls" />

> </If>

Author: Couldn't these `<If>` blocks be joined instead of having two separate ones? @manovotny

Reviewer: They absolutely could, but it felt a bit weird to enclose half the previous section with a new section. Ya know?

Happy to combine though. Just explaining why I didn't.

Reviewer: We discussed further in Slack. Going to leave this as is. 

Bridging of two sections in a single `<If />` is awkward. Better to keep them separate.
